src/cpp/redis.cpp

```diff
@@ -285,6 +285,59 @@ CommandReply Redis::set_model(const std::string& model_name,
     return run(cmd);
 }
 
+// Set a model from std::string_view buffer in the
+// database for future execution in a multi-GPU system
+void Redis::set_model_multigpu(const std::string& name,
+                               const std::string_view& model,
+                               const std::string& backend,
+                               int num_gpus,
+                               int batch_size,
+                               int min_batch_size,
+                               const std::string& tag,
+                               const std::vector<std::string>& inputs,
+                               const std::vector<std::string>& outputs)
+{
+    // Store a copy of the model for each GPU
+    for (int i = 0; i < num_gpus; i++) {
+        // Build the command
+        SingleKeyCommand cmd;
+        std::string device = "GPU:" + std::to_string(i);
+        std::string model_key = name + "." + device;
+        cmd << "AI.MODELSTORE" << Keyfield(model_key) << backend << device;
+
+        // Add optional fields if requested
+        if (tag.size() > 0) {
```

Comment:
I don't find this one compelling. Let me know if you feel strongly and we can discuss

Reply:
I would tend to agree with @mellis13 on this one. It seems like it'll make future maintainability easier since there's only one place we have to touch to construct these common areas of command construction